src/components/RecordTypesSelect/components/RecordItem.js

```diff
@@ -1,56 +1,111 @@
+  Dropdown,
+} from '@folio/stripes/components';
+
+import { IncomingRecordMenu } from './IncomingRecordMenu';
+import { IncomingRecordTrigger } from './IncomingRecordTrigger';
 
 import css from '../RecordTypesSelect.css';
 
 export const RecordItem = memo(({
   item,
   className,
   style,
-  onClick = noop,
+  onClick,
+  isEditable,
 }) => {
-  const {
-    type,
-    captionId,
-    iconKey,
-  } = item;
   const ref = useRef();
+  const [recordSelectorOpen, setRecordSelectorOpen] = useState(false);
+  const [incomingRecord, setIncomingRecord] = useState(item);
+
+  useEffect(() => {
+    setIncomingRecord(item);
+  }, [item]);
+
+  const isInitial = !className?.includes('incomingRecord');
 
-  const isClickable = onClick !== noop;
+  const trigger = triggerProps => (
+    <IncomingRecordTrigger
+      className={className}
+      style={style}
+      onClick={() => setRecordSelectorOpen(!recordSelectorOpen)}
+      iconKey={incomingRecord.iconKey}
+      captionId={incomingRecord.captionId}
+      {...triggerProps}
+    />
+  );
+  const menu = menuProps => (
+    <IncomingRecordMenu
+      onClick={record => {
+        setIncomingRecord(record);
+        onClick(record);
+        setRecordSelectorOpen(!recordSelectorOpen);
+      }}
+      {...menuProps}
+    />
+  );
 
-  return (
+  const initialButton = (
     <div // eslint-disable-line jsx-a11y/click-events-have-key-events
       data-test-record-item
       tabIndex="0"
       role="button"
-      id={type}
-      className={classNames(css.item, { [css.clickableItem]: isClickable }, className)}
+      id={incomingRecord.type}
+      className={classNames(css.item, { [css.clickableItem]: isEditable }, className)}
       style={style}
       ref={ref}
-      onClick={() => (isClickable ? onClick(item) : noop)}
+      onClick={() => (isEditable ? onClick(incomingRecord) : noop)}
     >
-      <AppIcon
-        size="medium"
-        app="data-import"
-        iconKey={iconKey}
-      >
-        <FormattedMessage id={captionId} />
-      </AppIcon>
+      {incomingRecord.iconKey ? (
+        <AppIcon
+          size="medium"
+          app="data-import"
+          iconKey={incomingRecord.iconKey}
+        >
+          <FormattedMessage id={incomingRecord.captionId} />
+        </AppIcon>
+      ) : (
+        <FormattedMessage id={incomingRecord.captionId} />
+      )}
+
+    </div>
+  );
+  const dropdownButton = (
+    <div data-test-incoming-record-selector>
+      <Dropdown
+        id="record-selector-dropdown"
+        open={recordSelectorOpen}
+        renderTrigger={trigger}
+        renderMenu={menu}
+        style={{ width: '100%' }}
+      />
     </div>
   );
+
+  return isEditable && !isInitial ? dropdownButton : initialButton;
 });
 
 RecordItem.propTypes = {
   item: PropTypes.object.isRequired,
   className: PropTypes.string,
   style: PropTypes.object,
   onClick: PropTypes.func,
+  isEditable: PropTypes.bool,
+};
+
+RecordItem.defaultProps = {
+  onClick: noop,
```

Comment:
It is unnecessary prop, cause in ` src/components/RecordTypesSelect/components/CompareRecordSelect.js` in this component you throw either function or noop.
Please recheck can this prop be omit